js/modules/k6/http/request.go

```diff
@@ -120,19 +124,24 @@ func (h *HTTP) Request(ctx context.Context, method string, url goja.Value, args
 //TODO break this function up
 //nolint: gocyclo
 func (h *HTTP) parseRequest(
-	ctx context.Context, method string, reqURL httpext.URL, body interface{}, params goja.Value,
+	ctx context.Context, method string, reqURL goja.Value, body interface{}, params goja.Value,
 ) (*httpext.ParsedHTTPRequest, error) {
 	rt := common.GetRuntime(ctx)
 	state := lib.GetState(ctx)
 	if state == nil {
 		return nil, ErrHTTPForbiddenInInitContext
 	}
 
+	u, err := httpext.ToURL(reqURL.Export())
```

Comment:
Hmm I think `goja.Runtime.ToValue()` is a relatively expensive operation. We'd probably be better off if `reqURL` was of an `interface{}` type here, and we had a check like this here:


```suggestion
	if urlJSValue, ok := reqURL.(goja.Value); ok {
    	reqURL = urlJSValue.Export()
	}
	u, err := httpext.ToURL(reqURL)
```

This way we won't have to covert the url from a JS value to a Go one, and back to a JS value, and back to a go value.

Reply:
Probably true .... but given this `common.Bind` code (among other):
https://github.com/k6io/k6/blob/da5eebc638d9917cedfe98e232a7eed4cb69a157/js/common/bridge.go#L231-L236
I am not certain ... how much of a saving this will be :shrug:

Reply:
Resolved by 39a0ed9f.